### PR TITLE
Allow PDF files for the thumbnail generation

### PR DESCRIPTION
**Pull request by ajay-gupta**
| Q | A
| --- | ---
| New feature? | yes
| BC breaks? | no

We can allow PDF files as well for the thumbnail generation which does not require any additional code changes in the bundle.

Please preview and merge.

### CODE REVIEW

File: Imagine/Data/DataManager.php [hunk 1]
```diff
@@ -149,8 +149,8 @@ public function find($filter, $path)
             throw new LogicException(sprintf('The mime type of image %s was not guessed.', $path));
         }
 
-        if (0 !== mb_strpos($binary->getMimeType(), 'image/')) {
-            throw new LogicException(sprintf('The mime type of image %s must be image/xxx got %s.', $path, $binary->getMimeType()));
+        if (0 !== mb_strpos($binary->getMimeType(), 'image/') && $binary->getMimeType() <> 'application/pdf') {
```
Comment (dbu) [inline Imagine/Data/DataManager.php:152]:
```suggestion
        if (0 !== mb_strpos($binary->getMimeType(), 'image/') && $binary->getMimeType() !== 'application/pdf') {
```

while `<>` seems to work too, i think !== is much more commonly used. is there a particular reason to use `<>` here?

**Review by dbu — COMMENTED** [review]
if everything else works with pdf, we should indeed allow that here, thanks for the PR. i have one question

File: Imagine/Data/DataManager.php (hunk 1 above)
Comment (ajay-gupta) [reply Imagine/Data/DataManager.php:152]: As there's no difference and absolutely no technical reason to use one over the other. For consistency we can switch to most commonly used operator. 
Thanks

**Comment by dbu** [comment]
can you please adjust the test DataManagerTest::testThrowsIfMimeTypeNotImageOneOnFind and testThrowsIfLoaderReturnBinaryWithMimeTypeNotImageOneOnFind to the new exception message? (see travis-ci build)

**Comment by ajay-gupta** [comment]
> 
> 
> can you please adjust the test DataManagerTest::testThrowsIfMimeTypeNotImageOneOnFind and testThrowsIfLoaderReturnBinaryWithMimeTypeNotImageOneOnFind to the new exception message? (see travis-ci build)

Is it only changes in exception message or something more? Please guide.

**Comment by dbu** [comment]
yes, just the expected message. you can find the phpunit output by clicking on "Details" in the `continous-integration/travis-ci/pr` line, which leads you e.g. to https://travis-ci.org/github/liip/LiipImagineBundle/jobs/727345430 . ignore the warning and the skipped tests, what is left is the two DataManagerTest failures.

**Comment by ajay-gupta** [comment]
Please review.

**Comment by dbu** [comment]
thanks a lot for this improvement!

**Comment by Seb33300** [comment]
https://github.com/liip/LiipImagineBundle/pull/1122...

**Comment by garak** [comment]
How is pdf thumbnail is supposed to work?
I tried in a project of mine to generate pdf thumbnails without changing code, but it doesn't work. It generates a pdf in my cache file (not an image)

**Comment by ajay-gupta** [comment]
It generates file of mime type - image only (file extension will not change).  It is working and rendering correctly for example:  
``

**Comment by garak** [comment]
Solved by forcing format in configuration. Thank you!

**Comment by Seb33300** [comment]
> How is pdf thumbnail is supposed to work? I tried in a project of mine to generate pdf thumbnails without changing code, but it doesn't work. It generates a pdf in my cache file (not an image)

I have the same issue, the generated thumbnail file is a PDF (not only the extension .pdf, if I edit the file with notepad I can see it starts by `%PDF-1.4`).
Is there any way to force the format only for PDF?

**Pull request merged**